### ISSUE TITLE
Opening a new patch doesn't refresh screen

### ISSUE THREAD

**Issue by metaphorz** [VCVRack/Rack#272]
OS (if applicable): OSX11 (on an iMac)
Version (or "dev" if compiling from source): 0.4.0

Whenever I open any patch after launching Rack on my iMac:

1) The screen is not refreshed with the new patch
2) Either I get a pop-up window stating that I do not have certain modules (which I do), OR
nothing happens
3) I exit from Rack
4) I re-launch Rack
5) The patch that was supposed to open is now open, and all modules are there. Everything works fine.

I need to exit Rack and relaunch whenever I change (or open) a new patch.

**Comment by metaphorz** [comment]
I have more information if it helps: I was just playing a patch. Let's call this A.
Then, I clicked "File->Open" to open another patch B. The result is that 

* B started playing to replace the sound of A
* The visual interface still shows A rather than B
* The drop-down menu for File is stuck on Open/Cmd+O

**Comment by AndrewBelt** [comment]
Does the display become unresponsive after choosing Open from the file select dialog?

**Comment by metaphorz** [comment]
Yes, the display is completely unresponsive as far as I can tell.

**Comment by metaphorz** [comment]
I re-downloaded VCV_rack on the upstairs iMac. For the MacBook, I didn't even need to do this -- things seem to be working fine now on both. I wonder if Rack auto-installs the latest version? I cannot see any other explanation on the Macbook working without me having to reinstall Rack.

Anyway- I think we can close this item. Thanks! Whatever happened seems to no longer result in  a frozen screen.

**Comment by AndrewBelt** [comment]
I'll leave it open because I suspect the issue is in the osdialog library (which I wrote a while ago).

**Comment by metaphorz** [comment]
Sounds good. For some reason, the iMac at work (separate from the other two I tried at home) still exhibits the above problem. I'll dig some more...

**Comment by metaphorz** [comment]
Thankfully, it has not happened since, but I will be on the lookout for it.

**Comment by metaphorz** [comment]
It started happening again on my work machine, which I had not used in a while. For the other two Macs (iMac at home and MacBook Pro), it has not happened since I reported the bug. I moved the Rack window and did not see any messages or message box. Here is what the screen looks like if I drag it to the bottom and back up:
 
![screenshot 2017-11-07 11 15 53_preview](https://user-images.githubusercontent.com/3519631/32507516-3fc03bce-c3ad-11e7-93f0-fbf7807603a1.png)

**Comment by metaphorz** [comment]
I just realized that I am running Yosemite on the affected machine. Let me update this machien to Sierra or High Sierra and try again.

**Comment by metaphorz** [comment]
Transitioned from Yosemite to High Sierra. The strange graphics problems seem to have disappeared.

**Comment by AndrewBelt** [comment]
Interesting, that almost confirms it's a driver issue, so perhaps the thing to do is to add a setting or flag to launch in opengl3 mode instead of opengl2. Not sure.

**Comment by cmountvpr** [comment]
This is happening to me too. Open a new patch, I hear the new patch, but still see the old one. Quit, relaunch, and the new one is there. I'm way back in Mountain Lion (!).

**Comment by metaphorz** [comment]
I think Mountain Lion and Yosemite probably have video drivers that are not as recent as Sierra or High Sierra. have you tried upgrading?

**Comment by cmountvpr** [comment]
It's a thought, but I'm not inclined to upgrade my OS for the sake of such a minor inconvenience.
